File: src/main/java/com/kamikazejam/datastore/mode/object/store/ObjectStorageDatabase.java
package com.kamikazejam.datastore.mode.object.store;
import com.kamikazejam.datastore.base.Cache;
import com.kamikazejam.datastore.base.storage.StorageDatabase;
import com.kamikazejam.datastore.mode.object.StoreObject;
import org.jetbrains.annotations.NotNull;

public class ObjectStorageDatabase<X extends StoreObject<X>> extends StorageDatabase<String, X> {

    public ObjectStorageDatabase(Cache<String, X> cache) {
        super(cache);
    }

    @Override
    public @NotNull String getLayerName() {
        return "Object MongoDB";
    }
}

File: src/main/kotlin/com/kamikazejam/datastore/mode/object/store/ObjectStorageDatabase.kt
package com.kamikazejam.datastore.mode.`object`.store

import com.kamikazejam.datastore.base.Collection
import com.kamikazejam.datastore.base.storage.StorageDatabase
import com.kamikazejam.datastore.mode.`object`.StoreObject

@Suppress("unused")
class ObjectStorageDatabase<X : StoreObject<X>>(collection: Collection<String, X>) :
    StorageDatabase<String, X>(collection) {
    override val layerName: String
        get() = "Object MongoDB"
}
